packages/databricks-sdk-js/src/config/KnownConfigLoader.ts

```diff
@@ -49,3 +49,44 @@ export class KnownConfigLoader implements Loader {
         cfg.attributes.resolveFromStringMap(iniFile[profile]);
     }
 }
+
+/**
+ * The ini library encodes dots as nested objects, which is not what we want.
+ * See: https://github.com/npm/ini/issues/22
+ *
+ * Reduce the nested object back to a flat object by concatenating object keys with a `.`
+ */
+function flattenIniObject(obj: {[key: string]: any}): {
+    [key: string]: any;
+} {
+    function _flattenIniObject(
+        obj: {
+            [key: string]: any;
+        },
+        topLevel: Array<string> = [],
+        resp = {}
+    ) {
+        const props: Record<string, any> = {};
+        for (const key in obj) {
+            if (typeof obj[key] === "object") {
+                topLevel.push(key);
+                resp = {
+                    ...resp,
+                    ..._flattenIniObject(obj[key], topLevel, resp),
+                };
+            } else {
+                props[key] = obj[key];
+            }
+        }
```

Comment:
This will reduce all nested objects to dots. But we only want to flatten objects  that
1. Start from direct children of the root ini object
2. Form a single chain of nested objects with no branches

Eg for the file
```ini
[a.b.c]
d = something

[one]
d = something
[.two]
e = something
```

We want to flatten the first object but for the second one we should rather throw an error and let the user know that it is an invalid cfg.

Reply:
This is actually what I want. I just want to treat the dot as a regular character. Added a unit test to document my intentions here.